tools/block-generator/generator/make_transactions.go

```diff
@@ -92,6 +115,43 @@ func (g *generator) makeAssetTransferTxn(header transactions.Header, receiver ba
 	}
 }
 
-func (g *generator) makeAssetAcceptanceTxn(header transactions.Header, index uint64) transactions.Transaction {
+func (g *generator) makeAssetAcceptanceTxn(header txn.Header, index uint64) txn.Transaction {
 	return g.makeAssetTransferTxn(header, header.Sender, 0, basics.Address{}, index)
 }
+
+// ---- application transactions ----
+
+func (g *generator) makeAppCreateTxn(sender basics.Address, round, intra uint64, approval, clear string) txn.Transaction {
+
+	createTxn := g.makeTestTxn(sender, round, intra)
+
+	/* all 0 values but keep around for reference
+	createTxn.ApplicationID = 0
+	createTxn.ApplicationArgs = nil
+	createTxn.Accounts = nil
+	createTxn.ForeignApps = nil
+	createTxn.ForeignAssets = nil
+	createTxn.Boxes = nil
+	createTxn.ExtraProgramPages = 0
+	*/
+
+	createTxn.Type = protocol.ApplicationCallTx
+	createTxn.ApprovalProgram = approval
+	createTxn.ClearStateProgram = clear
+
+	// sender opts-in to their own created app
+	createTxn.OnCompletion = txn.OptInOC
+
+	// max out local/global state usage but split
+	// 50% between bytes/uint64
+	createTxn.LocalStateSchema = basics.StateSchema{
+		NumUint:      8,
+		NumByteSlice: 8,
+	}
+	createTxn.GlobalStateSchema = basics.StateSchema{
+		NumUint:      32,
+		NumByteSlice: 32,
+	}
```

Comment:
global state is not currently utilized, but keep this optionality going forward.

Reply:
It's used in the next PR.